nifi-commons/nifi-expression-language/src/test/java/org/apache/nifi/attribute/expression/language/TestStandardPreparedQuery.java

```diff
@@ -309,6 +309,24 @@ public void testVariableImpacted() {
         assertTrue(Query.prepare("${anyMatchingAttribute('a.*'):equals('hello')}").getVariableImpact().isImpacted("attr"));
     }
 
+    @Test
+    public void testIsExpressionLanguagePresent() {
+        assertFalse(Query.prepare("value").isExpressionLanguagePresent());
+        assertFalse(Query.prepare("").isExpressionLanguagePresent());
+
+        assertTrue(Query.prepare("${variable}").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("${hostname()}").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("${hostname():equals('localhost')}").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("prefix-${hostname()}").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("${hostname()}-suffix").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("${variable1}${hostname()}${variable2}").isExpressionLanguagePresent());
+        assertTrue(Query.prepare("${${variable}}").isExpressionLanguagePresent());
+
+        assertFalse(Query.prepare("${}").isExpressionLanguagePresent());
+
+        assertTrue(Query.prepare("#{param}").isExpressionLanguagePresent());
```

Comment:
This should be false, not true. This is not expression language but rather a parameter reference. The value that will be made available to the component will be the fully resolved value, after substituting in the parameter.

Reply:
The parameter may or may not contain EL. My idea was to handle it as "unknown" and I felt it safer to say `true` in this case.

I ran into this "isExpressionLanguagePresent always returns true" issue in `customValidate()` and the parameter reference is already resolved when `customValidate()` gets called. So only `StringLiteralExpression` (non-EL) and `CompiledExpression` (EL) can occur there but no `ParameterExpression`. I was uncertain about `ParameterExpression` because I do not know where else it is being used in context of EL.

If you confirm that returning `false` is the right way I can easily change the logic. I just wanted to explain what my approach was.

Thanks

Reply:
Well, that's interesting. The user guide says:
```
Value - The value that will be used when the Parameter is referenced. Parameter values do not support Expression Language or embedded parameter references.
```

Parameters are not supposed to be able to make use of EL. But I just tried it with UpdateAttribute, and interestingly it did evaluate the reference. I think that is actually a bug, though. So I would tend to say that we should return `false` since Parameters are not supposed to support EL.

Reply:
Thanks @markap14. Changed the logic, also rebased the branch.